pandas/tests/extension/base/casting.py

```diff
@@ -32,11 +32,13 @@ def test_tolist(self, data):
         expected = list(data)
         assert result == expected
 
+    @pytest.mark.skip(reason="Floating precision issues")
```

Comment:
What's the reason for this skip? Because the conversion to string is different for numpy scalars vs native scalars? 

(the skip should also not be here, I think, but only for those dtypes that are failing them)

Reply:
Yes I think so. It's specifically that in the construction of expected
``` python
expected = pd.Series([str(x) for x in data[:5]], dtype=str)
```
we get a lot more decimal places. The astype operation works as before

Reply:
And what if you remove the `str(..)` call around `x`? Because by specifying string dtype, we will already use our internal machinery to handle the conversion, which should do it consistently for numpy vs native numbers, I think

Reply:
Tried this, problem persists (for both tests)

Reply:
Is this worth opening an issue about?

Reply:
> And what if you remove the `str(..)` call around `x`? Because by specifying string dtype, we will already use our internal machinery to handle the conversion, which should do it consistently for numpy vs native numbers, I think

Actually it did work when I used `data[:5]` directly versus the list comp :) Thanks!

Reply:
Hmm, but that's not really testing anymore what it was originally testing? Now the result and expected are created quite similarly (not necessarily using an independent code path). 

I think we should dive deeper to understand what is going on here. Do you have a small reproducer? 
For example in the following case, I don't see any difference between numpy.float64 scalar and python float in conversion to string:

```
In [26]: arr = np.random.randn(3)

In [27]: pd.Series([str(a) for a in arr], dtype=str)
Out[27]: 
0    0.29297099628905326
1     -1.409445837171125
2     1.0533994900019892
dtype: object

In [28]: pd.Series([str(a.item()) for a in arr], dtype=str)
Out[28]: 
0    0.29297099628905326
1     -1.409445837171125
2     1.0533994900019892
dtype: object
```

Reply:
Looking into this. It's surprisingly hard to get a copy-pastable reproducer but the test failure is reliable - will keep looking

Reply:
took a while but I finally got a copy-pastable example:
``` python
In [8]: import numpy as np
   ...: import pandas as pd
   ...: 
   ...: data = pd.array([0, 0.1, 0.2, 0.3, 0.4, 0.5], dtype="Float32")
   ...: s = pd.Series(data)
   ...: res = s.astype(str)
   ...: expected = pd.Series([x for x in data], dtype=str)

In [9]: expected
Out[9]: 
0                    0.0
1    0.10000000149011612
2    0.20000000298023224
3    0.30000001192092896
4     0.4000000059604645
5                    0.5
dtype: object
```

Reply:
It's only the 32 bit that fails for me locally (on a 64-bit system) - I wonder if it's something to do with that

Reply:
It seems to be a problem with `.item()` with `np.float32` scalar:
``` python
In [15]: scalar = np.float32(0.1)

In [16]: scalar
Out[16]: 0.1

In [17]: scalar.item()
Out[17]: 0.10000000149011612
```

Reply:
This was resolved by the numpy folks at https://github.com/numpy/numpy/issues/17880 (not a bug)